numpy/core/numeric.py

```diff
@@ -623,6 +624,45 @@ def asfortranarray(a, dtype=None):
     """
     return array(a, dtype, copy=False, order='F', ndmin=1)
 
+def pack_fields(a):
+    """Return copy of a structured array with padding between fields removed.
```

Comment:
And fields reordered in memory

Reply:
Oh right I forgot about reordering. Will think about it tomorrow.